src/transport/TransportMgrBase.cpp

```diff
@@ -14,44 +14,41 @@
  *    See the License for the specific language governing permissions and
  */
 
-/**
- * @file
- *   This file implements a stateless TransportMgr, it will took a raw message
- * buffer from transports, and then extract the message header without decode it.
- * For secure messages, it will pass it to the SecureSessionMgr, and for unsecure
- * messages (rendezvous messages), it will pass it to RendezvousSession.
- *   When sending messages, it will encode the packet header, and pass it to the
- * transports.
- *   The whole process is fully stateless.
- */
+#include <transport/TransportMgrBase.h>
 
+#include <support/CodeUtils.h>
 #include <transport/TransportMgr.h>
-
-#include <transport/RendezvousSession.h>
-#include <transport/SecureSessionMgr.h>
 #include <transport/raw/Base.h>
-#include <transport/raw/MessageHeader.h>
-#include <transport/raw/PeerAddress.h>
 
 namespace chip {
 
+CHIP_ERROR TransportMgrBase::SendMessage(const PacketHeader & header, const Transport::PeerAddress & address,
+                                         System::PacketBufferHandle && msgBuf)
+{
+    return mTransport->SendMessage(header, address, std::move(msgBuf));
+}
+
+void TransportMgrBase::Disconnect(const Transport::PeerAddress & address)
+{
+    mTransport->Disconnect(address);
+}
+
 CHIP_ERROR TransportMgrBase::Init(Transport::Base * transport)
 {
     if (mTransport != nullptr)
     {
         return CHIP_ERROR_INCORRECT_STATE;
     }
     mTransport = transport;
-    mTransport->SetMessageReceiveHandler(HandleMessageReceived, this);
+    mTransport->SetDelegate(this);
     ChipLogDetail(Inet, "TransportMgr initialized");
     return CHIP_NO_ERROR;
 }
 
 void TransportMgrBase::HandleMessageReceived(const PacketHeader & packetHeader, const Transport::PeerAddress & peerAddress,
-                                             System::PacketBufferHandle msg, TransportMgrBase * dispatcher)
+                                             System::PacketBufferHandle && msg)
```

Comment:
This function should take `msg` by value, not rvalue reference, because it doesn't always hand it off to someone else.

Reply:
I think it is fine to pass by rv ref. If the rv ref is silently dropped, the move-constructor is not called, the the outside destructor will free the packet.

But it will add complexity and confusion, I'll change it according the suggestion.

Reply:
Yes, you don't _leak_, of course.  But you get a situation where the outside function can access the data, or not, depending on the codepath taken in the inside function.  It makes a class of mistakes possible (examining the data after the call and having it work only sometimes) that can be avoided if we pass by value.